maven-assembly-plugin support

Any idea how I can make this compatible with artifacts built by the `maven-assembly-plugin`

I notice in the generated `pom.xml.new-version` that the `<finalName>` in the project's assembly plugin doesn't get updated to the new version and the assembly plugin throws a `NullPointerException`

```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-assembly-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>single</goal>
                        </goals>
                        <configuration>
                            <finalName>${project.artifactId}-${project.version}</finalName>
                            <descriptors>
                                <descriptor>src/main/assembly/assembly.xml</descriptor>
                            </descriptors>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

produces

```
Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single (default) on project projectA: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single (default) on project projectA: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:3.0.0:single failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: java.lang.NullPointerException
	at org.apache.maven.project.artifact.AttachedArtifact.getVersion(AttachedArtifact.java:138)
	at org.apache.maven.artifact.DefaultArtifact.validateIdentity(DefaultArtifact.java:148)
	at org.apache.maven.artifact.DefaultArtifact.<init>(DefaultArtifact.java:123)
	at org.apache.maven.project.artifact.AttachedArtifact.<init>(AttachedArtifact.java:49)
	at org.apache.maven.project.DefaultMavenProjectHelper.attachArtifact(DefaultMavenProjectHelper.java:60)
	at org.apache.maven.plugins.assembly.mojos.AbstractAssemblyMojo.execute(AbstractAssemblyMojo.java:487)
	at org.apache.maven.plugins.assembly.mojos.SingleAssemblyMojo.execute(SingleAssemblyMojo.java:58)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 21 more
```

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,4 @@
 .settings/
 target/
 **/.checkstyle
+*.iml

--- a/maven-external-version-plugin/src/main/java/org/apache/maven/version/ExternalVersionExtension.java
+++ b/maven-external-version-plugin/src/main/java/org/apache/maven/version/ExternalVersionExtension.java
@@ -22,6 +22,7 @@ package org.apache.maven.version;
 import org.apache.maven.AbstractMavenLifecycleParticipant;
 import org.apache.maven.MavenExecutionException;
 import org.apache.maven.artifact.ArtifactUtils;
+import org.apache.maven.artifact.versioning.VersionRange;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.model.Model;
 import org.apache.maven.model.Plugin;
@@ -114,6 +115,8 @@ public class ExternalVersionExtension
                 // now that we have the new version update the project.
                 mavenProject.setVersion( newVersion );
                 mavenProject.getArtifact().setVersion( newVersion );
+                // MNG-6641: setVersion nullifies versionRange
+                mavenProject.getArtifact().setVersionRange( VersionRange.createFromVersion( newVersion ) );
 
                 // TODO: get the unfiltered string, and re-filter it with new version.
                 String oldFinalName = mavenProject.getBuild().getFinalName();

